vtex/faststore
# Feat: Enable custom position of images in Image Gallery

## What's the purpose of this pull request?

This PR enables the customization of the image position in ImageGallery, as related in https://github.com/vtex/faststore/issues/1894

## How it works?

Now we have a new prop called `imagePos` that changes the `object-position` behavior of the images in ImageGallery.

```
<ImageGallery.Component
  data-fs-product-details-gallery
  {...ImageGallery.props}
  imagePosition="center"
  images={productImages}
/>
```


- Maps `imagePosition` prop in CMS:
<img width="578" alt="image" src="https://github.com/vtex/faststore/assets/3356699/41774d0d-f9f8-4736-84b9-2557cbc269b7">

- Renames the  `ImageZoom` component  to `ImageGalleryViewer` to avoid confusing users (Zoom feature not implemented yet)

## How to test it?

Change the `imagePosition` prop to either `top`, `center` or `bottom` to see the product image behavior in Product Details.

File: https://github.com/vtex/faststore/pull/1990/files#diff-9f0046ed9dc5dd5e1b9adbfbd15c83b52736882165c4a8f4533c895a248d621fR184

### [Starters Deploy Preview](https://starter-ly2ygldua-faststore.vercel.app/)

## References

https://developer.mozilla.org/en-US/docs/Web/CSS/object-position
